lib/dradis/plugins/burp/formats/xml.rb

```diff
@@ -0,0 +1,125 @@
+module Dradis::Plugins::Burp::Formats
+
+  # This module knows how to parse Burp XML format.
+  module Xml
+    private
+
+    def import_xml(file_content)
+      logger.info { 'Parsing Burp Scanner XML output file...' }
+      doc = Nokogiri::XML(file_content)
+      logger.info { 'Done.' }
+
+      if doc.root.name != 'issues'
+        error = "Document doesn't seem to be in the Burp Scanner XML format."
+        logger.fatal { error }
+        content_service.create_note text: error
+        return false
+      end
+
+      # This will be filled in by the Processor while iterating over the issues
+      @hosts         = []
+      @affected_host = nil
+      @issue_text    = nil
+      @evidence_text = nil
+
+      doc.xpath('issues/issue').each do |xml_issue|
+        process_issue(xml_issue)
+      end
+
+      logger.info { 'Burp Scanner results successfully imported' }
+      true
+    end
+
+    # Creates the Nodes/properties
+    def process_issue(xml_issue)
+      host_label = xml_issue.at('host')['ip']
+      host_label = xml_issue.at('host').text if host_label.empty?
+      affected_host = content_service.create_node(label: host_label, type: :host)
+      logger.info { "\taffects: #{host_label}" }
+
+      unless @hosts.include?(affected_host.label)
+        @hosts << affected_host.label
+        url = xml_issue.at('host').text
+        affected_host.set_property(:hostname, url)
+        affected_host.save
+      end
+
+      # Burp extensions don't follow the "unique type for every Issue" logic
+      # so we have to deal with them separately
+      if xml_issue.at('type').text.to_str == '134217728'
```

Comment:
What does this number mean? I might be good to save this value in a semantic variable. Ex: `BURP_TYPE_ID = 134217728` (replace BURP_TYPE_ID with what it actually is)